File: tests/benchmarks/cpp/02_fibonacci_recursive.cpp
// ベンチマーク2: フィボナッチ数列（再帰版）
// 30番目のフィボナッチ数を計算

#include <chrono>
#include <iostream>

using namespace std;
using namespace std::chrono;

int fibonacci(int n) {
    if (n <= 1)
        return n;
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main() {
    int n = 30;

    auto start = high_resolution_clock::now();
    int result = fibonacci(n);
    auto end = high_resolution_clock::now();

    auto duration = duration_cast<microseconds>(end - start);

    cout << "Fibonacci(" << n << ") = " << result << endl;
    cout << "Time: " << duration.count() / 1000000.0 << " seconds" << endl;

    return 0;
}

File: tests/benchmarks/rust/02_fibonacci_recursive.rs
// ベンチマーク2: フィボナッチ数列（再帰版）
// 30番目のフィボナッチ数を計算

use std::time::Instant;

fn fibonacci(n: i32) -> i32 {
    if n <= 1 {
        return n;
    }
    fibonacci(n - 1) + fibonacci(n - 2)
}

fn main() {
    let n = 30;

    let start = Instant::now();
    let result = fibonacci(n);
    let duration = start.elapsed();

    println!("Fibonacci({}) = {}", n, result);
    println!("Time: {:.6} seconds", duration.as_secs_f64());
}